clang/lib/Parse/ParseDecl.cpp

```diff
@@ -4456,6 +4456,12 @@ void Parser::ParseDeclarationSpecifiers(
       isInvalid = DS.setFunctionSpecNoreturn(Loc, PrevSpec, DiagID);
       break;
 
+    case tok::kw__Export:
+      // If we find kw__Export, it is being applied to a var or function
+      // This will be handled in ParseDeclaratorInternal()
```

Comment:
If this is a declaration specifier, shouldn't this work like any other one? e.g. https://godbolt.org/z/s4Tz1vjn4 (it's an abomination, to be sure, but it's odd to claim this as a declaration specifier but then not implement it as one).

Reply:
The _Export keyword is part of the declarator.  This code is to stop the declspec parsing.  For example, in `int _Export x,y;` only `x` is exported.  The flag was added to the DeclSpec class to handle the case of `class _Export A`.  That is later passed on to the Decl for the class.  If there is a better way to do this, I'd be interested.  We do have to match the existing XL compiler for backwards compatibility.

Reply:
> in int _Export x,y; only x is exported.

That is deeply strange behavior IMO. I can't think of anything else in the language that works like that in that syntactic position.

> We do have to match the existing XL compiler for backwards compatibility.

That can be handled in the downstream too. For example, in this case, I would expect a diagnostic to let the user know that `_Export` very likely does not behave the way they expect in your code example.

Reply:
> That is deeply strange behavior IMO. I can't think of anything else in the language that works like that in that syntactic position.

In the traditional implementation, `_Export` is a declarator like `*` is, except it binds more strongly than `(<function parameters>)` or `[<array bound>]`.